### ISSUE TITLE
Typo (?) in inference/tracker.py leads to ConfigAttributeError

### ISSUE THREAD

**Issue by YiLin-Zhou** [talmolab/dreem#86]
## Bug description

Running `dreem-train` on sleap data throws a `ConfigAttributeError: Key 'model' is not in struct`.

## Fix

Changing line 197  `curr_track += 1`  in dreem/inference/tracker.py to the following:
```
curr_track_id += 1
```
fixed this issue. Given the surrounding lines all reference ` curr_track_id` , I think it was a typo?

## Logs

```
Error executing job with overrides: []
Traceback (most recent call last):
  File "C:\Users\ylzhou\dreem\dreem\training\train.py", line 55, in run
    model = train_cfg.get_model()
            ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\dreem\dreem\io\config.py", line 94, in get_model
    model_params = self.cfg.model
                   ^^^^^^^^^^^^^^
omegaconf.errors.ConfigAttributeError: Key 'model' is not in struct
    full_key: model
    object_type=dict
```

**Comment by aaprasad** [comment]
Hi @YiLin-Zhou, I'm not really sure how that would fix this error, considering `dreem-train` doesn't call anything in `dreem.inference`... This error more likely came from not having your config set up properly initially

**Comment by YiLin-Zhou** [comment]
You're right, I think I frankenstein-ed the wrong error log & bug fix into the same error request. The logs / bug description I posted above is indeed due to a config error, but there is a real bug too, fixed by changing the line to `curr_track_id += 1`.

This is the (real) logs for it from running dreem-track:
```
Error executing job with overrides: []
Traceback (most recent call last):
  File "C:\Users\ylzhou\dreem\dreem\inference\track.py", line 131, in run
    preds = track(model, trainer, dataloader)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\dreem\dreem\inference\track.py", line 72, in track
    preds = trainer.predict(model, dataloader)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\pytorch_lightning\trainer\trainer.py", line 858, in predict
    return call._call_and_handle_interrupt(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\pytorch_lightning\trainer\call.py", line 47, in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\pytorch_lightning\trainer\trainer.py", line 897, in _predict_impl
    results = self._run(model, ckpt_path=ckpt_path)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\pytorch_lightning\trainer\trainer.py", line 981, in _run
    results = self._run_stage()
              ^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\pytorch_lightning\trainer\trainer.py", line 1020, in _run_stage
    return self.predict_loop.run()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\pytorch_lightning\loops\utilities.py", line 178, in _decorator
    return loop_run(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\pytorch_lightning\loops\prediction_loop.py", line 124, in run
    self._predict_step(batch, batch_idx, dataloader_idx, dataloader_iter)
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\pytorch_lightning\loops\prediction_loop.py", line 253, in _predict_step
    predictions = call._call_strategy_hook(trainer, "predict_step", *step_args)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\pytorch_lightning\trainer\call.py", line 319, in _call_strategy_hook
    output = fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\miniconda3\envs\dreem\Lib\site-packages\pytorch_lightning\strategies\strategy.py", line 437, in predict_step
    return self.lightning_module.predict_step(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\dreem\dreem\models\gtr_runner.py", line 177, in predict_step
    frames_pred = self.tracker(self.model, batch[0])
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\dreem\dreem\inference\tracker.py", line 77, in __call__
    return self.track(model, frames)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\dreem\dreem\inference\tracker.py", line 141, in track
    instances_pred = self.sliding_inference(model, frames)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ylzhou\dreem\dreem\inference\tracker.py", line 197, in sliding_inference
    curr_track += 1
    ^^^^^^^^^^
UnboundLocalError: cannot access local variable 'curr_track' where it is not associated with a value

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
Predicting DataLoader 0:   0%|                                                                   | 0/1 [00:04<?, ?it/s]
```

Sorry for the confusion!

**Comment by aaprasad** [comment]
@YiLin-Zhou this issue should be patched as of #87. Pull from main to get the latest version. I'm going to close this issue but feel free to open a new one if you find anything else!
